### ISSUE TITLE
indeterminate progressbar is broken on every kvantum theme

### ISSUE THREAD

**Issue by Thenujan-0** [tsujan/Kvantum#852]
Indeterminate progressbar looks like this in every kvantum theme and it doesn't move.
![image](https://user-images.githubusercontent.com/85059778/215989634-684e2f26-b5ae-4ac6-8f51-260c8b6f5834.png)

But it looks fine on other application styles

**Comment by Thenujan-0** [comment]
Sorry for not being specific enough. This screenshot is from kirigami gallery.
The progressbar is `QQC2.ProgressBar{indeterminate:true}`
https://develop.kde.org/docs/use/kirigami/components-progressbars/

Is this actually completed or did you close this because i didn't give enough info?

**Comment by tsujan** [comment]
Oh, that's QtQuick, not QWidget. Kvantum is a QWidget style, not a QtQuick style. Please read [Theme-Making.pdf](https://github.com/tsujan/Kvantum/blob/master/Kvantum/doc/Theme-Making.pdf) → A word about QML/Qt Quick.

KDE's `qqc2-desktop-style` is responsible for such problems. It tries to apply the widget style to QtQuick apps, and that's an inefficient approach.

In this case, there is no "indeterminate progressbar" in [QProgressBar](https://doc.qt.io/qt-6/qprogressbar.html). `qqc2-desktop-style` gives a null progresbar to [CE_ProgressBarContents](https://doc.qt.io/qt-6/qstyle.html#ControlElement-enum), and Kvantum lets [QCommonStyle](https://doc.qt.io/qt-6/qcommonstyle.html) draw it.

In short, what you see is what `QCommonStyle` does with the poor information `qqc2-desktop-style` gives to it. Kvantum doesn't draw anything in this case.

> Is this actually completed or did you close this because i didn't give enough info?

I closed it because it isn't a Kvantum bug.

However, I'll try to find a short workaround. Thanks!

**Comment by tsujan** [comment]
I made the animation of busy progressbars independent of the existence of `QWidget`: https://github.com/tsujan/Kvantum/commit/fa386ccc85e0df92fb99076727fe1ab432b8c11f

If you can compile and install Kvantum from the latest git source, please check whether it works with QtQuick progressbars (I have no QML app here).

**Comment by Thenujan-0** [comment]
>If you can compile and install Kvantum from the latest git source, please check whether it works with QtQuick progressbars (I have no QML app here).


Works like a charm :smile:  thank you so much :heart:

**Comment by tsujan** [comment]
Thank *you*, for telling me about this QtQuick exception and also for testing!
